doc/application/coccinelle.rst

```diff
@@ -0,0 +1,509 @@
+.. _coccinelle:
+
+..
+   Copyright 2010 Nicolas Palix <npalix@diku.dk>
+   Copyright 2010 Julia Lawall <julia.lawall@lip6.fr>
+   Copyright 2010 Gilles Muller <Gilles.Muller@lip6.fr>
+
+Coccinelle
+##########
+
+Coccinelle is a tool for pattern matching and text transformation that has
+many uses in kernel development, including the application of complex,
+tree-wide patches and detection of problematic programming patterns.
+
+.. note::
+   Linux and macOS development environments are supported, but not Windows.
+
+Getting Coccinelle
+******************
+
+The semantic patches included in the kernel use features and options
+which are provided by Coccinelle version 1.0.0-rc11 and above.
+Using earlier versions will fail as the option names used by
+the Coccinelle files and ``coccicheck`` have been updated.
+
+Coccinelle is available through the package manager
+of many distributions, e.g. :
+
+.. hlist::
+   * Debian
```

Comment:
Sorry, missed one. Needs to be a blank line before the body of this hlist directive.  Otherwise LGTM.